src/tests/multihost/ad/test_access_control.py

```diff
@@ -12,6 +13,23 @@
 from sssd.testlib.common.exceptions import SSHLoginException
 
 
+def ssh_login(multihost, username):
+    client_hostname = multihost.client[0].sys_hostname
+    client = pexpect_ssh(client_hostname, username, 'Secret123', debug=False)
+    try:
+        client.login()
+    except SSHLoginException:
+        pytest.fail(f'{username} failed to login')
```

Comment:
If You put the fail here, You can not use the function for negative scenario where You check that the user can not login.
I would put result of the function call in the assert instead.

Reply:
This log in will be due to password incorrect, The exception to handle 'access denied' is added with 'err'

Reply:
Yes, You can not use the function to cover this negative scenario if You fail the test inside a helper.
The test itself should be explicitly checking/asseriing the expected return value from the helper function.